opm/parser/eclipse/Parser/Parser.hpp

```diff
@@ -101,6 +102,29 @@ namespace Opm {
             addParserKeyword( std::unique_ptr< ParserKeyword >( new T ) );
         }
 
+        // below are some helper functions
+
+        static EclipseState parse(std::shared_ptr<const Deck> deck, const ParseContext& context = ParseContext());
```

Comment:
I am not sure, but I can try.  It would imply that EclipseGrid can be constructed from a reference.  I don't think it is impossible, but I don't know how far that would propagate.  Investigating now.